### PR TITLE
(TELCO-574) block charm when NRF goes down

### PR DESCRIPTION
**Pull request by Gmerold**
# Description

This PR adds handling of the NRF relation broken event. 
If NRF will ever go down, the AMF charm will reflect that fact by going into `Blocked` state.

# Checklist:

- [x] My code follows the [style guidelines](/CONTRIBUTING.md) of this project
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that validate the behaviour of the software
- [x] I validated that new and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have bumped the version of the library

### CODE REVIEW

File: test-requirements.txt [hunk 1]
```diff
@@ -1,4 +1,4 @@
-juju==3.2.0.0
+juju==3.2.0.1
```
Comment (gatici) [inline test-requirements.txt:1]: Do we need that specific version ? Why are we binding to that minor version ?

File: tests/integration/test_integration.py [hunk 2]
```diff
@@ -79,3 +74,28 @@ async def test_relate_and_wait_for_active_status(
         status="active",
         timeout=1000,
     )
+
+
+@pytest.mark.abort_on_fail
+async def test_remove_nrf_relation_and_wait_for_blocked_status(
```
Comment (gatici) [inline tests/integration/test_integration.py:80]: I think  test_remove_nrf_and_wait_for_blocked_status as test name makes more sense as we are removing NRF. Consequently NRF relation is removed. 

We can add another test just removing the relation and while keeping NRF deployed.

**Review by saltiyazan — APPROVED** [review]

File: test-requirements.txt (hunk 1 above)
Comment (Gmerold) [reply test-requirements.txt]: Good question :) There was probably some bug in the lib, which made us pin the version. 
Not needed now.
Removed.

File: tests/integration/test_integration.py (hunk 2 above)
Comment (Gmerold) [reply tests/integration/test_integration.py]: Updated.

**Review by gatici — APPROVED** [review]

**Pull request merged**